WorldWideWeb/NXShims.h

```diff
@@ -0,0 +1,85 @@
+//
+//  NXShims.h
+//  WorldWideWeb
+//
+//  Created by Chris Brandow on 8/23/24.
+//
+
+#ifndef NXShims_h
+#define NXShims_h
+
+
+#endif /* NXShims_h */
+
+/// These are largely just copied from Next Manual
+/// http://www.bitsavers.org/pdf/next/Release_1_Dec90/NEXTstep_Reference_Volume_1_Dec90.pdf
+/// http://www.bitsavers.org/pdf/next/Release_1_Dec90/NEXTstep_Reference_Volume_2_Dec90.pdf
+
+typedef struct NXRect {
+    NXPoint origin;
+    NXSize size;
+} NXRect;
+
+
+typedef struct NXTextBlock {
+    struct _NXTextBlock *next; /* Next block in linked list */
+    struct _NXTextBlock *prior; /* Previous block in linked list */
+    struct _bFlags {
+        unsigned int malloced:1; /* True if block was malloced */
+        unsigned int PAD: 15;
+    } tbFlags;
+    short chars; /* Number of characters in block */
+    unsigned char *text; /* The text */
+} NXTextBlock;
+
+/* Describes tabstop. */
+typedef struct _NXTabStop {
+    short kind; /* Only NX_LEFTTAB implemented */
+    NXCoord x; /* x coordinate for stop */
```

Comment:
Fun fact: `NXCoord` actually still appears in the present-day IOKit API: https://developer.apple.com/documentation/iokit/nxcoord

I believe the modern equivalent is `CGFloat`, can we use that?